src/task.h

```diff
@@ -149,13 +154,11 @@ task_deregister_thread(task_system *ts,
 
 
 platform_status
-task_system_create(platform_heap_id    hid,
-                   platform_io_handle *ioh,
-                   task_system       **system,
-                   bool                use_stats,
-                   bool                use_bg_threads,
-                   uint8               num_bg_threads[NUM_TASK_TYPES],
-                   uint64              scratch_size);
+task_system_create(platform_heap_id          hid,
+                   platform_io_handle       *ioh,
+                   task_system             **system,
+                   const task_system_config *cfg,
+                   uint64                    scratch_size);
```

Comment:
Maybe make scratch_size part of config?

Reply:
Should we scrap the scratches? It's bending over backwards to reduce `malloc`s, but it's awkward and also since we do it for foreground threads, it means that threads that only read still consume a bunch of memory for no reason.

Reply:
Moved scratch into config for now, but it will go away in a subsequent PR.